src/components/ImageUpload.tsx

```diff
@@ -1,66 +1,130 @@
+import { Image } from "cloudinary-react";
+import { useCallback, useState } from "react";
 import { useDropzone } from "react-dropzone";
-import { theme } from "./../theme/theme";
-import { StyledPaperInput } from "./StyledComponents";
+import { StyledFilledButton, StyledPaperInput } from "./StyledComponents";
+
+// once we have backend authentication we will need to protect these variables
+const PUBLIC_CLOUDINARY_CLOUD_NAME = "projectnewm",
+  PUBLIC_CLOUDINARY_UPLOAD_PRESET = "rvktckuk";
 
 interface ImageUploadProps {
-  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
-  errors: FormikErrors<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: string;
-      type: string;
-    };
-  }>;
-  touched: FormikTouched<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: null;
-      type: string;
-    };
-    releaseDate: string;
-    title: string;
-    yourRole: string;
-  }>;
+  fileType?: "png" | "jpg" | "jpeg" | "wdp" | "jp2" | "bmp" | "pdf" | "tiff" | "ico" | "eps";
+  dropzoneLabel?: string;
+  buttonLabel?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFieldValue?: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  setTouched?: (fields: { [field: string]: boolean }, shouldValidate?: boolean) => void;
 }
 
-export const ImageUpload = (props: ImageUploadProps) => {
-  const { setFieldValue } = props;
+export const ImageUpload = ({
+  fileType = "png",
+  dropzoneLabel = "Drag & Drop (Square Image Only)",
+  buttonLabel = "Upload Image",
+  setFieldValue,
+  setTouched,
+}: ImageUploadProps) => {
+  const [imgLoaded, setImgLoaded] = useState<boolean>(false);
+  const [uploadedFilePublicId, setUploadedFilePublicId] = useState<string>();
+  const onDrop = useCallback(
+    acceptedFiles => {
+      const url = `https://api.cloudinary.com/v1_1/${PUBLIC_CLOUDINARY_CLOUD_NAME}/upload`;
+
+      acceptedFiles.forEach(async (acceptedFile: File) => {
+        const formData = new FormData();
+        formData.append("file", acceptedFile);
+        formData.append("upload_preset", String(PUBLIC_CLOUDINARY_UPLOAD_PRESET));
+
+        const response = await fetch(url, {
+          body: formData,
+          method: "post",
+        });
+
+        const data = await response.json();
+        setUploadedFilePublicId(data.public_id);
+        setFieldValue && setFieldValue("uploadedImageId", data.public_id, true);
+        setImgLoaded(false);
+      });
+    },
+    [setFieldValue]
+  );
+
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     accept: "image/*",
-    onDrop: acceptedFiles => {
-      setFieldValue("files", acceptedFiles);
-    },
+    multiple: false,
+    onDrop,
   });
-  return (
-    <Box>
-      { }
+  let render = <div />;
+  if (!uploadedFilePublicId) {
+    render = (
```

Comment:
Wondering if the render variable is needed.
I think we should be able to just return early and drop the else

```kotlin
if(!uploadedFilePublicId) {
    return (
        ....
    )
}

//if we got here, uploadedFilePublicId is true
return (...)
```

Reply:
Changed